### ISSUE TITLE
aws_eip data provider does not support Classic EIPs

### ISSUE THREAD

**Issue by Ashex** [terraform-providers/terraform-provider-aws#2825]
The following snippet leads to a crash when the EIP is on ec2-classic. It does work if it is a VPC EIP:

### Terraform Version
Terraform v0.11.1
+ provider.aws v1.6.0

### Affected Resource(s)
- aws_eip

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "aws" {
    version = "1.6"
}

data "aws_eip" "this" {
  public_ip = "10.10.10.10"
}
```

### Debug Output
https://gist.github.com/Ashex/7035dc2d335dfc00650495e59816b979

### Panic Output
https://gist.github.com/Ashex/7035dc2d335dfc00650495e59816b979

### Expected Behavior
EIP allocation Id is retriever

### Actual Behavior
Terraform crashes with a EOF error

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
Verified that this occurs on aws provider 1.5.0

When I run this with a VPC EIP it succeeds, this appears to only fail with a Classic EIP

**Comment by micemuni** [comment]
Not only classic IP. We are not able to attach the EIP that created manually. then i am trying to attach to instance creating via terraform

**Comment by bflad** [comment]
Hi folks 👋 Very sorry for the lengthy delays with merging this support, there were a few conflicting pull requests to the same code which needed to get sorted out. Good news is that the following will be supported in version 1.44.0 of the AWS provider, likely releasing later today or tomorrow:
* Search by `filter`
* Search be `tags`
* EC2-Classic support

**Comment by bflad** [comment]
This has been released in [version 1.44.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1440-november-14-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
